presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java

```diff
@@ -193,14 +193,28 @@ public String getPartitionName()
                 s3SelectPushdownEnabled && S3SelectPushdown.isCompressionCodecSupported(inputFormat, path)));
     }
 
-    private static void checkBlocks(List<InternalHiveBlock> blocks, long start, long length)
+    private static void checkBlocks(Path path, List<InternalHiveBlock> blocks, long start, long length)
     {
-        checkArgument(length >= 0);
-        checkArgument(!blocks.isEmpty());
-        checkArgument(start == blocks.get(0).getStart());
-        checkArgument(start + length == blocks.get(blocks.size() - 1).getEnd());
+        checkArgument(length >= 0, "Split size is negative for %s, split from %s with length %s", path, start, length);
+        checkArgument(!blocks.isEmpty(), "No blocks for %s for split from from %s with length %s", path, start, length);
+        long blocksStartPosition = blocks.get(0).getStart();
+        checkArgument(start == blocksStartPosition, "Unexpected blocks start position for %s. Expected %s, but got %s", path, start, blocksStartPosition);
+        long blocksEndPosition = blocks.get(blocks.size() - 1).getEnd();
+        long end = start + length;
+        checkArgument(end == blocksEndPosition,  "Unexpected blocks end position for %s. Expected %s, but got %s", path, end, blocksEndPosition);
         for (int i = 1; i < blocks.size(); i++) {
-            checkArgument(blocks.get(i - 1).getEnd() == blocks.get(i).getStart());
+            long currentBlockStart = blocks.get(i).getStart();
+            long previousBlockEnd = blocks.get(i - 1).getEnd();
+            checkArgument(
+                    previousBlockEnd == currentBlockStart,
+                    "Blocks are not continuous for %s with split that start %s with length %s. Block %s (out of %s) starts at %s, while previous block ends at %s",
```

Comment:
`contiguous`

```suggestion
                    "Blocks are not contiguous for %s for split start=%s, length=%s. Block %s (out of %s) starts at %s, while previous block ends at %s",
```

Reply:
However, i think in this case, and also in `!(start == blocksStartPosition)`, `!(end == blocksEndPosition)` cases, it would be better to just dump all the start,end of all blocks. Eg when check fails at first or second block, maybe we could see if further blocks are reasonable or not.